sympy/tensor/array/sparse_ndim_array.py

```diff
@@ -109,7 +109,24 @@ def reshape(self, *newshape):
         if new_total_size != self._loop_size:
             raise ValueError("Invalid reshape parameters " + newshape)
 
-        return type(self)(*(newshape + (self._array,)))
+        return type(self)(*(newshape + (self._sparse_array,)))
+    
+    def _eval_derivative_n_times(self, s, n):
```

Comment:
maybe you should not overwrite this specific method. Try to look for `_visit_eval_derivative_<...>` methods.

Reply:
I believe that is it `al_derivative_array` that should be overridden.